/*
 * Copyright 2013 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Author: slamm@google.com (Stephen Lamm)

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package net_instaweb;


// Protobuf of CSS resolved for each <link> tag when a page initially loads.
message CriticalCssResult {
  message LinkRules {
    // The unrewritten URL of the <link> tag that loaded the critical rules.
    optional string link_url = 1;
    // The unrewritten critical CSS rules (selectors and declarations)
    // for one <link> tag. If the string is empty, it means none of the
    // rules were critical.
    optional string critical_rules = 2;
    optional int32 original_size = 3;
  };
  repeated LinkRules link_rules = 1;
  optional int32 exception_count = 2;
  optional int32 import_count = 3;
  optional int32 link_count = 4;
}
